src/compat/db_json.cpp

```diff
@@ -137,6 +157,47 @@ static void db_json_merge_two_json_arrays (JSON_DOC &array1, const JSON_DOC *arr
 static void db_json_merge_two_json_by_array_wrapping (JSON_DOC &j1, const JSON_DOC *j2);
 static void db_json_copy_doc (JSON_DOC &dest, const JSON_DOC *src);
 
+static void db_json_get_paths_helper (const JSON_VALUE &obj, const std::string &sql_path,
+				      std::vector<std::string> &paths);
+static void db_json_normalize_path (std::string &path_string);
+static bool db_json_isspace (const unsigned char &ch);
+static int db_json_convert_pointer_to_sql_path (const char *pointer_path, const JSON_DOC &doc,
+    std::string &sql_path_out, bool &path_has_effect);
+static int db_json_convert_sql_path_to_pointer (const char *sql_path, const JSON_DOC &doc,
+    std::string &json_pointer_out);
+static JSON_PATH_TYPE db_json_get_path_type (std::string &path_string);
+STATIC_INLINE void db_json_build_path_special_chars (const JSON_PATH_TYPE &json_path_type,
+    std::unordered_map<std::string, std::string> &special_chars);
+STATIC_INLINE std::vector<std::string> db_json_split_string_by_delimiter (const std::string &s,
+    const std::string &delim);
+STATIC_INLINE void db_json_replace_token_special_chars (std::string &token,
+    const std::unordered_map<std::string, std::string> &special_chars);
+
+STATIC_INLINE JSON_VALUE *db_json_doc_to_value (const JSON_DOC *doc);
+
+JSON_POINTER::JSON_POINTER (const char *pointer_path) : rapidjson::GenericPointer<JSON_VALUE> (pointer_path),
+  pointer_path_string (pointer_path) {};
+
+const JSON_POINTER JSON_POINTER::CreateParent()
+{
+  std::string parent_pointer_path_string;
+  int path_length = pointer_path_string.length();
+  int i = path_length - 1;
+
+  for (i = path_length - 1; i >= 0; --i)
+    {
+      if (pointer_path_string[i] == '/')
+	{
+	  break;
+	}
+    }
+
+  parent_pointer_path_string = pointer_path_string.substr (0, i);
+  const JSON_POINTER pointer_parent (parent_pointer_path_string.c_str());
+
+  return pointer_parent;
```

Comment:
this function always creates the same parent, how about saving the parent in a variable so that further calls do not have to recalculate it? we could actually compute it in the constructor and always return that as a reference. doing so, we also avoid the extra copy (or compute it in CreateParent, but only once).

Reply:
we don't need to do this, because we will have a pointer inside each function which will point to a specific path not the entire json, so the CreateParent function does not act like a getter.